README
|  \/  |  __ _ __  __(_) _ __ ___   _   _  ___
| |\/| | / _` |\ \/ /| || '_ ` _ \ | | | |/ __|
| |  | || (_| | >  < | || | | | | || |_| |\__ \
|_|  |_| \__,_|/_/\_\|_||_| |_| |_| \__,_||___/

This is the web front- and backend for Maximus. Maximus is a BlitzMax module
manager.

maximus-web is responsible for keeping an up to date BlitzMax module repository.
Its main tasks consists of fetching modules and packing them to a useable archive
format for maximus, the commandline application.

GETTING STARTED

At http://maximus.htbaa.com you can find a running version of maximus-web. But
if you want to host your own, perhaps private, version of maximus-web you should
start with the INSTALL file.

You can also use Vagrant (http://vagrantup.com) to quickly setup a virtual
machine that will install all required software and configure Maximus. Once
ready just visit http://localhost:3000 to use the Maximus website.

Run script/maximus_server.pl to test the application.

LICENSE

This software is licensed under the MIT license. Please see the COPYING file.


Changes
Revision history for Maximus.

    - Added tmuxgo.sh script for quickly setting up a tmux session
    - Updated Vagrantfile for Vagrant 1.1
    - Added the module description to the meta data of the announcement message
    - Test for Twitter Broadcast driver now mocks Net::Twitter (gh-53)
    - Improved performance for processing download requests
    - Added a XML Sitemap (gh-137)
    - Updated INSTALL
    - Added examples for using SQLite or PostgreSQL in maximus.default.conf
    - Moved some search queries into the ResultSet namespace to clean up the
      controllers
    - Fixed table index names since PostgreSQL doesn't allow duplicates
    - Script::SQL::DDL now also generates a schema file for PostgreSQL
    - Updated My Account page as the "under development" message was heavily
      outdated
    - Improved documentation
    - Minimal required Perl is now 5.12

1.004001  2013-04-20
    - Now assuming that the remote_location of an archive is a full url (gh-144)

1.004000  2013-02-03
    - Added a doc/pod browser for Vagrant users, connect at
      http://localhost:3002. Documentation is regenerated every time vagrant
      provision has run
    - Now setting HTTP cache headers for sources file and timeline feeds
    - Loading PageCache plugin last as advised
    - SCM settings for Git can now have their main branch specified (gh-118)
    - Added FTP and FileSystem transport (gh-135)
    - Authors are now listed on the /authors page (gh-90)
    - Converted JavaScript to CoffeeScript

1.003002  2012-09-17
    - Fixed bug in BlitzMax lexer (gh-133)

1.003001  2012-09-15
    - Typefix

1.003000  2012-09-15
    - Added Flattr button and added extra support buttons on client page
    - Module download links now point to remote_location if available. This
      goes for both the module page and sources file (gh-122)
    - Added a Mojolicious::Lite script (script/maximus_mojo.pl) that can be
      used for only serving module archives
    - Removed compatibility for Internet Explorer 6 (gh-97)
    - Now also storing module history and displaying it on the module page if
      it's available (gh-89, gh-111)
    - Revamped module page to show the module license and author(s) if
      applicable (gh-111)
    - Improved lexer to also retrieve module license and author(s). This data
      is now being stored in the module version database table (gh-111)
    - Fixed detection of module version and description (gh-126)
    - Fixed assigning modules when adding a SCM configuration (gh-125)
    - Deleting a SCM configuration no longer deletes all linked modules (gh-124)
    - Generating new DDL has been made easier by simply looking at the Schema
      module version
    - Only show ReCAPTCHA when a public key is present
    - Upped minimal required version of Perl from 5.10.1 to 5.12.1
    - Module search include now works properly by using forward() instead of the
      Catalyst::View::Component::SubInclude plugin (gh-123)
    - Menu item 'modules' is highlighted again (gh-120)
    - Updated Catalyst::Runtime to 5.90015
    - Fixed maximus_testing.conf and maximus.default.conf since empty elements
      are no longer allowed
    - Added Vagrant support for quickly setting up a development environment
      that's close to the actual deployment situation. Once booted simply direct
      your browser to http://localhost:3000 to visit the Maximus website
    - When using Windows it's now expected that Git is accessible from the
      command line and has been set in PATH

1.002002  2012-04-29
    - Forgot to update link to MooTools-Core 1.4.5
    - Removed Log::Log4perl::Catalyst dependency, it's provided by Log::Log4perl

1.002001  2012-04-29
    - Added Pledgie button to the Client page
    - Added screencast to the Client page
    - Added screenshots to the Client page
    - Updated Catalyst::Plugin::RequireSSL to 0.07 (gh-116)
    - Updated MooTools-Core to 1.4.5

1.002000  2012-01-14
    - Fixed a bug in the algorithm that searches for module dependencies. When
      subdirectories included files the algorithm would fail (gh-115)
    - Added basic module search functionality (gh-33)
    - When using MySQL please take over the extra settings from the default
      config file: on_connect_call and on_connect_do
    - Added tests for Form roles (gh-72)
    - maximus_worker.pl and maximus_task.pl now have an assigned process name,
      set by $0 (gh-109)
    - Added reCaptcha at signup form to prevent bot signups (gh-110)
    - Replaced Head JS with LABjs (gh-112)

1.001000  2011-12-12
    - Module description no longer required. If omitted it'll try to detect the
      description from the module source file. This is both the case for manual
      archive uploads, auto-discovery and through SCM updating
    - Fixed link to module management from SCM management page
    - Added ability to do a forced updated for a SCM configuration (gh-85)
    - Cache::FileCache namespace set to Maximus 
    - Now showing statistics about modules (gh-87)
    - Updated MooTools-Core to 1.4.2
    - Updated MooTools-More to 1.4.0.1
    - Upgraded to Catalyst 5.9
    - Switched Catalyst::Log::Log4perl for Log::Log4perl::Catalyst (gh-106)

1.000005  2011-08-09
    - .o files are no longer excluded when generating archives

1.000004  2011-08-01
    - Added client_version to the configuration file which is being used by the
      client page. This allows for updating the latest client release on the
      website without having to go through modifying templates and skipping the
      need for tagging a new release when only the client has been updated

1.000003  2011-07-17
    - Updated client page for client 1.0.0 release

1.000002  2011-07-09
    - Added client download link to Linux binary and Windows installer

1.000001  2011-07-08
    - Now using nfreeze to store data for Gearman jobs so they can be read on
      by workers on servers with a different architecture

1.000000  2011-06-28

Makefile.PL
#!/usr/bin/env perl -w
use inc::Module::Install;

name 'Maximus';
all_from 'lib/Maximus.pm';

requires 'Archive::Extract'                             => '0.38';
requires 'Archive::Zip'                                 => '1.30';
requires 'Catalyst::Runtime'                            => '5.90015';
requires 'Catalyst::Model::DBIC::Schema'                => '0.43';
requires 'Catalyst::Action::RenderView'                 => '0.14';
requires 'Catalyst::Authentication::Store::DBIx::Class' => '0.1300';
requires 'Catalyst::Model::Adaptor'                     => '0.08';
requires 'Catalyst::Plugin::Assets'                     => '0.036';
requires 'Catalyst::Plugin::Authentication'             => '0.10018';
requires 'Catalyst::Plugin::Authorization::Roles'       => '0.08';
requires 'Catalyst::Plugin::Cache'                      => '0.10';
requires 'Catalyst::Plugin::Compress::Zlib'             => '0.01';
requires 'Catalyst::Plugin::ConfigLoader'               => '0.28';
requires 'Catalyst::Plugin::PageCache'                  => '0.22';
requires 'Catalyst::Plugin::Session'                    => '0.30';
requires 'Catalyst::Plugin::Session::Store::DBIC'       => '0.12';
requires 'Catalyst::Plugin::Session::State::Cookie'     => '0.17';
requires 'Catalyst::Plugin::StackTrace'                 => '0.11';
requires 'Catalyst::Plugin::Static::Simple'             => '0.29';
requires 'Catalyst::Plugin::RequireSSL'                 => '0.07';
requires 'Catalyst::TraitFor::Controller::reCAPTCHA'    => '1.112960';
requires 'Catalyst::View::Email'                        => '0.30';
requires 'Catalyst::View::TT'                           => '0.34';
requires 'Catalyst::View::XML::Feed'                    => '0.03';
requires 'Cache::Cache'                                 => '1.06';
requires 'Config::General'                              => '2.50';
requires 'CSS::Minifier::XS'                            => '0.05';
requires 'Data::SearchEngine'                           => '0.32';
requires 'DateTime'                                     => '0.61';
requires 'DateTime::Format::MySQL'                      => '0.04';
requires 'DBD::mysql'                                   => '4.016';
requires 'DBI'                                          => '1.613';
requires 'DBIx::Class'                                  => '0.08127';
requires 'DBIx::Class::Schema::Loader'                  => '0.07010';
requires 'Digest::SHA'                                  => '5.48';
requires 'Email::Valid'                                 => '0.184';
requires 'Exception::Class'                             => '1.30';
requires 'File::Copy::Recursive'                        => '0.38';
requires 'File::Path'                                   => '2.08';
requires 'File::Slurp'                                  => '9999.13';
requires 'File::Temp'                                   => '0.22';
requires 'Gearman::Client'                              => '1.11';
requires 'Gearman::Server'                              => '1.11';
requires 'HTML::FormHandler'                            => '0.34001';
requires 'HOP::Lexer'                                   => '0.032';
requires 'JavaScript::Minifier::XS'                     => '0.07';
requires 'JSON::Any'                                    => '1.22';
requires 'local::lib'                                   => '1.006005';
requires 'Log::Dispatch'                                => '2.26';
requires 'Log::Log4perl'                                => '1.28';
requires 'File::Type'                                   => '0.22';
requires 'Moose'                                        => '2.0007';
requires 'MooseX::NonMoose'                             => '0.20';
requires 'MooseX::Params::Validate'                     => '0.14';
requires 'MooseX::SemiAffordanceAccessor'               => '0.08';
requires 'MooseX::Storage'                              => '0.25';
requires 'namespace::autoclean'                         => '0.11';
requires 'Net::Twitter'                                 => '3.17001';
requires 'Net::FTP::AutoReconnect'                      => '0.03';
requires 'Path::Class'                                  => '0.19';
requires 'SQL::Translator'                              => '0.11006';
requires 'Storable'                                     => '2.21';
requires 'Search::Sitemap'                              => '2.13';
requires 'Template'                                     => '2.22';
requires 'Template::Plugin::Digest::SHA1'               => '0.03';
requires 'Template::Plugin::Cache'                      => '0.13';
requires 'version'                                      => '0.82';
requires 'YAML'                                         => '0.71';
requires 'XML::Simple'                                  => '2.18';
test_requires 'Cache::Null'                             => '2.04';
test_requires 'DateTime::Format::SQLite'                => '0.11';
test_requires 'DBD::SQLite'                             => '1.29';
test_requires 'Test::More'                              => '0.88';
test_requires 'Test::Exception'                         => '0.31';
test_requires 'Test::MockObject'                        => '1.20120301';
test_requires 'Test::WWW::Mechanize::Catalyst'          => '0.56';
test_requires 'Net::FTP::Mock'                          => '0.103300';
recommends 'App::HTTPThis'                              => '0.002';
recommends 'FCGI'                                       => '0.71';
recommends 'FCGI::ProcManager'                          => '0.19';
recommends 'Plack'                                      => '0.9978';
recommends 'Plack::Middleware::ReverseProxy'            => '0.09';
recommends 'Starman'                                    => '0.2010';
recommends 'Mojolicious'                                => '3.20';
recommends 'Pod::Site'                                  => '0.53';
recommends 'DBD::Pg'                                    => '2.19.3';
recommends 'DateTime::Format::Pg'                       => '0.16008';
catalyst;
perl_version '5.012001';
license 'MIT';

requires_external_bin 'svn';
requires_external_bin 'git';

install_script glob('script/*.pl');
auto_install;
WriteAll;
